src/dotty/tools/dotc/core/Types.scala

```diff
@@ -2727,7 +2727,7 @@ object Types {
 
         case tp @ AnnotatedType(annot, underlying) =>
           val underlying1 = this(underlying)
-          if (underlying1 eq underlying) tp else underlying1
+          if (underlying1 eq underlying) tp else tp.derivedAnnotatedType(annot, underlying1)
```

Comment:
If `annot.tree` is a complex expression which again contains types, do we have to map over these as well?